shared/util.ts

```diff
@@ -1,7 +1,7 @@
 import { GameVariable, IUpgrade, IUpgradeEffect, VariableModFunction } from "./types";
 import { allUpgrades } from "./upgrades";
 
-function pluck<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
+export function pluck<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
```

Comment:
This is for my curiosity: does this method mutate the originally array (in this example `clickyBits`) by removing anything that satisfies the predicate? Then returns the last removed object? Typescript is still a bit new to me, just making sure I'm reading this method correctly.

Reply:
basically yes! now that I'm looking at it more critically, I'm realizing it actually removes all items that satisfy the predicate, but only returns one of the items (the last one). In the two current use cases, there should only ever be one item in the list that matches the predicate. Would probably be worth clarifying that or maybe fixing it to only remove the first item... anyone have thoughts about that?